### DISCUSSION TITLE
Is there a way to access `config` helper function in second argument of `plugin` function?

### DISCUSSION THREAD

**Discussion by saadeghi** [tailwindlabs/tailwindcss#12308] [Help]
[`plugin` functions](https://tailwindcss.com/docs/plugins#overview) has a `config()` helper function that allows us to read anything from Tailwind config file.

And plugin function can have [a second argument](https://tailwindcss.com/docs/plugins#providing-default-values) for default values.

So in this config, is there a way to have access to a custom config property (`mycolor`) inside the second argument of `plugin` function?  
https://play.tailwindcss.com/fiA2z9aRhs?file=config
```js
// tailwind.config.js
const plugin = require('tailwindcss/plugin')

module.exports = {
  // a custom property
  mycolor: 'blue',

  plugins: [
    plugin(

      function ({ config }) {
        // this works. it prints the `mycolor` value
        console.log(config('mycolor'))
      },
      {
        theme: {
          extend: {
            colors: {
              // this doesn't work. I need to use config() here
              cool: plugin(function ({ config }) {
                return config('mycolor')
              }),
            },
          },
        },
      }

    ),
  ],
}
```

**Comment by alex-krasikau** [comment]
Hello @saadeghi!

Sure, you can replace any child of the `theme` or `extend` property with function that receives a config object as an argument:

```
require("tailwindcss/plugin")(
  function ({config}) {
    console.log(config('mycolor'))
  },
  {
      theme: {
          extend: {
              colors: ({theme}) => {
                  console.log(theme("colors.red.500"))
                  return {}
              },
          },
      },
  }
),
```

**Reply by saadeghi** [reply]
It's not inside `theme`. It's top level, next to `theme`
```js
module.exports = {
  mycolor: 'blue',
  theme: {
    //...
  }
  //...
}
```
